core/src/test/java/org/apache/iceberg/rest/RESTCatalogAdapter.java

```diff
@@ -665,7 +670,7 @@ public static void configureResponseFromException(
   }
 
   private static Namespace namespaceFromPathVars(Map<String, String> pathVars) {
-    return RESTUtil.decodeNamespace(pathVars.get("namespace"));
+    return RESTUtil.decodeNamespace(pathVars.get("namespace"), NAMESPACE_SEPARATOR);
```

Comment:
I thought that @jackye1995 suggested sending the separator from the client each time. Is that not what we want to do here?

Reply:
yes and that is being handled in https://github.com/apache/iceberg/pull/10905. However, as a first step we need to make the namespace separator configurable (regardless of whether it's configurable from the server or the client), which is being handled in this PR. Making it controllable from the client and send a query param is handled after this PR in https://github.com/apache/iceberg/pull/10905 (and requires a vote on the [spec change](https://github.com/apache/iceberg/pull/10904))